src/moonlink/src/row/proto_converter.rs

```diff
@@ -0,0 +1,128 @@
+use crate::row::{MoonlinkRow, RowValue};
+use moonlink_proto::moonlink as MoonlinkProto;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ProtoToMoonlinkRowError {
```

Comment:
we should try put errors in one please... ow we have 3-4 layers of nested errors...